orchestration/api/instances.py

```diff
@@ -99,6 +98,26 @@ def instance_ops(tenant_id=''):
         # If description is not provided, the instance creation should proceed
         logger.info("no user_id provided. Exception [%s]", str(e))
 
+    # action of the instance creator
+    try:
+        action = content['action']
+        if action == '':
+            raise ValueError('Empty action provided')
+        if action == 'opensds.provision-volume':
```

Comment:
Why don't we pass all the prospective IPs to lower level and let lower level decide which to pick. Example, the python-script action knows that it is calling Hotpot API, then it will pick hotpot_ip else it can pick gelato_ip or xyz_ip